packages/compiler-cli/src/ngtsc/transform/src/compilation.ts

```diff
@@ -490,6 +491,28 @@ export class TraitCompiler implements ProgramTypeCheckAdapter {
     }
   }
 
+  extendedTemplateCheck(sf: ts.SourceFile, extendedTemplateChecker: ExtendedTemplateChecker):
+      ts.Diagnostic[] {
+    const classes = this.fileToClasses.get(sf);
+    if (classes === undefined) {
+      return [];
+    }
+
+    const diagnostics: ts.Diagnostic[] = [];
+    for (const clazz of classes) {
+      if (isNamedClassDeclaration(clazz)) {
```

Comment:
Invert this condition:
```suggestion
      if (!isNamedClassDeclaration(clazz)) {
        continue;
      }
```

This keeps the rest of the code cleaner - the "main case" should always be the least indented and guarded.